lua/nvim-treesitter/configs.lua

```diff
@@ -135,6 +135,9 @@ local function disable_all(mod)
   end
 
   disable_mod_conf_autocmd(mod)
+  if type(config_mod.enable) == 'table' then
+    config_mod.enabled_langs = config_mod.enable
+  end
```

Comment:
lets move the config value, to a separate value to protect it from state changes. This is not nice but is due to the fact that after this change `enable` will be a config and a state variable in user config but our code requires a separations of those concerns.

Reply:
Alternative would be to use `currently_enabled` for the state. This might be confusing for everyone relying on the old behavior and has to risk of confusing state and config in some places.

Reply:
We can not do that in setup or define_modules because we don't know which of the ones happens first and setup does not know which of the modules actually end up being modules.